daemon/src/maker.rs

```diff
@@ -23,19 +26,54 @@ mod wire;
 #[database("maker")]
 pub struct Db(sqlx::SqlitePool);
 
+#[derive(Clap)]
+struct Opts {
+    /// The port to listen on for p2p connections.
+    #[clap(long, default_value = "9999")]
+    p2p_port: u16,
+    /// The port to listen on for the HTTP API.
+    #[clap(long, default_value = "8001")]
+    http_port: u16,
+
+    /// URL to the electrum backend to use for the wallet.
+    #[clap(long, default_value = "ssl://electrum.blockstream.info:60002")]
+    electrum: String,
+
+    /// Where to permanently store data, defaults to the current working directory.
+    #[clap(long)]
+    data_dir: Option<PathBuf>,
+
+    /// Generate a seed file within the data directory.
+    #[clap(long)]
+    generate_seed: bool,
+}
+
 #[rocket::main]
 async fn main() -> Result<()> {
-    let client =
-        bdk::electrum_client::Client::new("ssl://electrum.blockstream.info:60002").unwrap();
+    let opts = Opts::parse();
+
+    let data_dir = opts
+        .data_dir
+        .unwrap_or_else(|| std::env::current_dir().expect("unable to get cwd"));
+
+    if !data_dir.exists() {
+        tokio::fs::create_dir_all(&data_dir).await?;
+    }
+
+    let seed = Seed::initialize(&data_dir.join("seed"), opts.generate_seed).await?;
+
+    let client = bdk::electrum_client::Client::new(&opts.electrum).unwrap();
 
     // TODO: Replace with sqlite once https://github.com/bitcoindevkit/bdk/pull/376 is merged.
-    let db = bdk::sled::open("/tmp/maker.db")?;
+    let db = bdk::sled::open(data_dir.join("wallet_db"))?;
```

Comment:
as per other comment:

```suggestion
    let db = bdk::sled::open(data_dir.join("taker_wallet_db"))?;
```